cylc/flow/task_remote_mgr.py

```diff
@@ -316,19 +316,18 @@ def _get_remote_tidy_targets(
             A mapping of install targets to valid platforms only where
             platforms are available.
         """
-        if not platform_names and install_targets:
+        if install_targets and not platform_names:
             install_targets_map: Dict[str, List[Dict[str, Any]]] = {
                 t: [] for t in install_targets}
+            unreachable_targets = install_targets
         else:
             install_targets_map = get_install_target_to_platforms_map(
                 platform_names, quiet=True)
+            # If we couldn't find a platform for a target, we cannot tidy it -
+            # raise an Error:
+            unreachable_targets = install_targets.difference(
+                install_targets_map)
```

Comment:
Ok, I'm happy with this as `get_install_target_to_platforms_map()` does not produce any empty values in the dictionary 👍